pkg/cmd/service/service.go

```diff
@@ -147,6 +148,29 @@ type Config struct {
 	Profiler   ProfilerConfig
 }
 
+func DefaultConfigWithRandomPorts() (*Config, error) {
+	config := DefaultConfig()
+
+	l, err := net.Listen("tcp", "")
+	if err != nil {
+		return nil, err
+	}
+	defer l.Close()
+	httpPort := l.Addr().(*net.TCPAddr).Port
+
+	l, err = net.Listen("tcp", "")
+	if err != nil {
+		return nil, err
+	}
+	defer l.Close()
+	grpcPort := l.Addr().(*net.TCPAddr).Port
+
+	config.GRPC.Addr = fmt.Sprintf(":%d", grpcPort)
```

Comment:
At this point do you think it is worth explicitly using [`netip.AddrPort`](https://pkg.go.dev/net/netip#AddrPort) for the type of `config.GRPC.Addr` (and also the HTTP version). I would prefer having a typed address rather than a string.

Reply:
Probably, yes. Let me look into that approach as an alternative and see how it turns out 👍